src/services/user-group/hooks/index.js

```diff
@@ -76,20 +79,55 @@ const deleteWholeClassFromCourse = (hook) => {
 	});
 };
 
+const courseInviteHook = async hook => {
+	if (hook.path === 'courses' && hook.params.query && hook.params.query.link) {
+		const dbLink = await hook.app.service('link').get(hook.params.query.link) // link is used as "authorization"
+		delete hook.params.query.link
+		if (dbLink) return restrictToCurrentSchool(hook)
+	}
+
+	return restrictToUsersOwnCourses(hook)
+}
+
+const courseInvitePatchHook = async hook => {
+	const subsequentHooks = [globalHooks.hasPermission('USERGROUP_EDIT'), restrictToCurrentSchool, globalHooks.permitGroupOperation,
+		deleteWholeClassFromCourse
+	]
+	if (!hook.params.query || !hook.params.query.link) return callSubsequentHooks()
+
+	const dbLink = await hook.app.service('link').get(hook.params.query.link) // link is used as "authorization"
+	delete hook.params.query.link
+	return callSubsequentHooks(!!dbLink)
+
+	async function callSubsequentHooks(skip) {
```

Comment:
The naming from skip do not give information about what is expecting.  With testUserGroupEditPermission=true/false or other naming it is more clarify. You must swich the condition two lines  lower. If you change it.